file/file_prefetch_buffer.cc

```diff
@@ -66,6 +66,17 @@ void FilePrefetchBuffer::CalculateOffsetAndLen(size_t alignment,
     // chunk_len is greater than 0.
     bufs_[index].buffer_.RefitTail(static_cast<size_t>(chunk_offset_in_buffer),
                                    static_cast<size_t>(chunk_len));
+  } else if (chunk_len > 0) {
```

Comment:
For async prefetching we don’t refit the data so it allocate new memory. Planning to handle this in next iteration

Reply:
I didn't understand this case. By the time we get here, PrefetchAsync would have already consumed as much data as it can from the buffers. IIUC, one or both buffers should be empty, i.e chunk_len will be 0. When exactly will this case be true?

Reply:
I will try to add the unit test to catch this error. But this was faced in db_stress